demos/imagenet_classification_demo/grid_mat.hpp

```diff
@@ -0,0 +1,176 @@
+                    bool isFpsTest, bool showAccuracy,
+                    Presenter& presenter) {
+        // Draw a rectangle
+        cv::Point p1 = cv::Point(0, 0);
+        cv::Point p2 = cv::Point(outImg.cols, rectangleHeight);
+        rectangle(outImg, p1, p2, cv::Scalar(0, 0, 0), cv::FILLED);
+
+        presenter.drawGraphs(outImg);
+        
+        // Show FPS
+        double fontScale = 1.5;
+        int thickness = 2;
+        int baseline = 0;
+        cv::Size textSize = cv::getTextSize("", cv::FONT_HERSHEY_PLAIN, fontScale, thickness, &baseline);
+        cv::Scalar textColor = cv::Scalar(255, 255, 255);
+        int textPadding = 10;
+
+        cv::putText(outImg,
+                    cv::format("FPS: %0.01f", avgFPS),
+                    cv::Point(textPadding, textSize.height + textPadding),
+                    cv::FONT_HERSHEY_PLAIN, fontScale, textColor, thickness);
+        cv::putText(outImg,
+                    cv::format("Latency: %dms", static_cast<int>(avgLatency * 1000)),
+                    cv::Point(textPadding, (textSize.height + textPadding) * 2),
+                    cv::FONT_HERSHEY_PLAIN, fontScale, textColor, thickness);
+        
+        if (showAccuracy) {
+            std::string accuracyMessage = "Accuracy (top 0): 0.000";
+            cv::Size accuracyMessageSize = cv::getTextSize(accuracyMessage,
+                                                       cv::FONT_HERSHEY_PLAIN, fontScale, thickness, &baseline);
+            cv::putText(outImg,
+                        cv::format("Accuracy (top %d): %.3f", FLAGS_nt, accuracy),
+                        cv::Point(outImg.cols - accuracyMessageSize.width - textPadding, textSize.height + textPadding),
+                        cv::FONT_HERSHEY_PLAIN, fontScale, textColor, thickness);
+        }
+        
+        std::string testMessage = "Testing, please wait...";
+        cv::Size testMessageSize = cv::getTextSize(testMessage,
+                                                   cv::FONT_HERSHEY_PLAIN, fontScale, thickness, &baseline);
+        if (isFpsTest) {
+            cv::putText(outImg,
+                        testMessage,
+                        cv::Point(outImg.cols - testMessageSize.width - textPadding,
+                                  (textSize.height + textPadding) * 2),
+                        cv::FONT_HERSHEY_PLAIN, fontScale, cv::Scalar(50, 50, 255), thickness);
+        }
+    }
+
+    cv::Mat getMat(std::vector<long long> &endTimes, bool showLabels) {
+        while (!updateList.empty()) {
+            cv::Mat frame = std::get<0>(updateList.front());
+            std::string predictedLabel = std::get<1>(updateList.front());
+            cv::Scalar textColor = (std::get<2>(updateList.front())
+                                                ? cv::Scalar(75, 255, 75) : cv::Scalar(50, 50, 255));
+            updateList.pop_front();
+
+            if (updateList.empty()) {
```

Comment:
The check should be removed, because most of processed images are not shown because of it.